src/synthesized_datasets/_dtypes.py

```diff
@@ -0,0 +1,58 @@
+from enum import Enum
+
+import pyspark.sql.types as st
+
+
+class DType(str, Enum):
+    """Class to define the internal dtypes that can be handled."""
+
+    BOOL = "bool"
+    NULLABLE_BOOL = "boolean"
+    DATETIME = "datetime64[ns]"
+    DATE = "date"
+    FLOAT = "float32"
+    DOUBLE = "float64"
+    INTEGER = "int32"
+    LONG = "int64"
+    NULLABLE_LONG = "Int64"
+    STRING = "string"
+    TIMEDELTA = "timedelta64[ns]"
```

Comment:
Why this subset of types? are there not more?
Nullable float, int are some that come to mind